executor/runcexecutor/executor.go

```diff
@@ -335,7 +335,11 @@ func (w *runcExecutor) Run(ctx context.Context, id string, root executor.Mount,
 
 	err = exitError(ctx, err)
 	if err != nil {
-		releaseContainer(context.TODO())
+		if rec != nil {
```

Comment:
nit: this block is just a duplicate of the `err == nil` case - I think we could avoid this by setting another error value?